### ISSUE TITLE
the result of an assignment expression is corrupted

### ISSUE THREAD

**Issue by drsm** [nginx/njs#52]
when [ToPropertyKey](https://tc39.github.io/ecma262/#sec-topropertykey) performs conversion of an object value:
```js
>> var x = {valueOf: function() { return 'val'; }, toString: function() { return {}; }};
undefined
>> var a = {};
undefined
>> var b = (a[x] = 'test');
undefined
>> a
{
 val: 'test'
}
>> b
{
 
}
```

```js
>> var aa = {};
undefined
>> var bb = (aa[{}] = 'test')
undefined
>> bb
'[object Object]'
```

**Comment by drsm** [comment]
@xeioex 

This seems fixed somewhere after `0.3.1`

**Comment by drsm** [comment]
```diff
diff -r 2f1f71863953 njs/test/njs_unit_test.c
--- a/njs/test/njs_unit_test.c  Sat Apr 27 09:04:24 2019 +0800
+++ b/njs/test/njs_unit_test.c  Sat Apr 27 23:27:47 2019 +0300
@@ -4975,6 +4975,14 @@
                  "          valueOf: function() { return 1 } };  a"),
       nxt_string("1") },
 
+    { nxt_string("var a = { toString: function() { return [] },"
+                 "          valueOf: function() { return 1 } };"
+                 "var o = {}; o[a] = 'test'"),
+      nxt_string("test") },
+
+    { nxt_string("({})[{}] = 'test'"),
+      nxt_string("test") },
+
     { nxt_string("var o = {b:$r.props.b}; o.b"),
       nxt_string("42") },
```
